### PR TITLE
Add project ancestor groups logic

### PR DESCRIPTION
**Pull request by BallyB**
Adding the logic to retrieve ancestor group(s) of a specific project. See https://docs.gitlab.com/ee/api/projects.html#list-a-projects-groups

### CODE REVIEW

File: NGitLab.Tests/ProjectsTests.cs [hunk 1]
```diff
@@ -735,4 +735,38 @@ public async Task CreateProjectWithSquashOption(SquashOption? inputSquashOption)
...
+        using var context = await GitLabTestContext.CreateAsync();
+        var projectClient = context.Client.Projects;
+        var group = context.CreateGroup();
+        var subgroup = context.CreateSubgroup(group.Id);
+        var project = context.CreateProject(subgroup.Id);
+
+        var groups = projectClient.GetGroupsAsync(project.Id, new ProjectGroupsQuery()).ToArray();
+
+        Assert.That(groups.Select(g => g.Id), Contains.Item(group.Id));
+        Assert.That(groups.Select(g => g.Id), Contains.Item(subgroup.Id));
```
Comment (louis-z) [inline NGitLab.Tests/ProjectsTests.cs]:
```suggestion
        Assert.That(groups.Select(g => g.Id), Is.EquivalentTo(new int[] { group.Id, subgroup.Id }));
```

File: NGitLab/Impl/ProjectClient.cs [hunk 2]
```diff
@@ -173,6 +175,38 @@ public GitLabCollectionResponse<Project> GetForksAsync(string id, ForkedProjectQ
...
+        {
+            url = Utils.AddParameter(url, "shared_min_access_level", (int)query.SharedMinAccessLevel);
+        }
+
+        if (query.SkipGroups is not null)
+        {
+            url = Utils.AddParameter(url, "skip_groups", query.SkipGroups);
+        }
+
+        url = Utils.AddParameter(url, "shared_visible_only", value: query.SharedVisibleOnly);
```
Comment (louis-z) [inline NGitLab/Impl/ProjectClient.cs]:
```suggestion
        url = Utils.AddParameter(url, "shared_visible_only", query.SharedVisibleOnly);
```

File: NGitLab/Models/ProjectGroupsQuery.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+    /// <summary>
+    /// Limit to shared groups with at least this access level
+    /// </summary>
+    public AccessLevel? SharedMinAccessLevel { get; set; }
+
+    /// <summary>
+    /// Skip the group IDs passed
+    /// </summary>
+    public int?[] SkipGroups { get; set; }
```
Comment (louis-z) [inline NGitLab/Models/ProjectGroupsQuery.cs]: Here it's the array that should be optional, not its items, right?

```suggestion
    public int[] SkipGroups { get; set; }
```

**Review by louis-z — CHANGES_REQUESTED** [review]

File: NGitLab/Models/ProjectGroupsQuery.cs (hunk 3 above)
Comment (BallyB) [reply NGitLab/Models/ProjectGroupsQuery.cs]: Yes, I couldnt set the entire array nullable so, but sure we can remove it

**Review by louis-z — APPROVED** [review]

**Pull request merged**